run_tests.py

```diff
@@ -422,6 +477,17 @@ def test_from_failstr(failstr):
 if state_reader.has_file and not test_resume.from_file:
     state_reader.drop()
 
+# filter testcases
+resume_filter = test_resume.filter()
+tests = [
+    t
+    for t in tests
+    for _ in range(int(n_count))
+    if resume_filter(t)
+    and (not inclusions or shell.testcase_in(t, inclusions))
+    and not shell.testcase_in(t, exclusions)
+]
+
```

Comment:
I understand it was presented in the code before  and not relate to the PR changes, but this code is very hard readable because of complexity.

Reply:
Yes, I agree, and I've reworked the cycle. Maybe you have another way?